sycl/source/kernel_bundle.cpp

```diff
@@ -290,12 +290,52 @@ std::vector<kernel_id> get_kernel_ids() {
 }
 
 bool is_compatible(const std::vector<kernel_id> &KernelIDs, const device &Dev) {
-  std::set<detail::RTDeviceBinaryImage *> BinImages =
-      detail::ProgramManager::getInstance().getRawDeviceImages(KernelIDs);
-  return std::all_of(BinImages.begin(), BinImages.end(),
-                     [&Dev](const detail::RTDeviceBinaryImage *Img) {
-                       return doesDevSupportDeviceRequirements(Dev, *Img);
-                     });
+  if (KernelIDs.empty())
+    return false;
+  // TODO: also need to check architectures matching
+  auto doesImageTargetMatchDevice = [](const device &Dev,
+                                       const detail::RTDeviceBinaryImage &Img) {
+    const char *Target = Img.getRawData().DeviceTargetSpec;
+    auto BE = Dev.get_backend();
+    if (strcmp(Target, __SYCL_PI_DEVICE_BINARY_TARGET_SPIRV64) == 0) {
+      return (BE == sycl::backend::opencl ||
```

Comment:
There is also `backend::ext_intel_esimd_emulator`, should it be also supported by `is_compatible`?

Reply:
7cce706

Reply:
Can you please add a test for that too?

Reply:
b4cb2dd

Reply:
Test for esimd: [187bf48](https://github.com/intel/llvm/pull/9769/commits/187bf48c79615cd1d6ffa53e382021cc32a391b1)